spec/compiler/semantic/case_spec.cr

```diff
@@ -244,4 +244,27 @@ describe "Semantic: case" do
       ),
       "warning in line 3\nWarning: case without condition must have an `else` clause."
   end
+
+  it "always requires an else for Flags enum" do
+    assert_warning %(
+        struct Enum
+          def includes?(other : self)
+            false
+          end
+        end
+
+        @[Flags]
+        enum Color
+          Red
+          Green
+          Blue
+        end
+
+        e = Color::Red
+        case e
+        when .red?
```

Comment:
Wouldn't this fail exhaustiveness even if `Color` wasn't `@[Flags]`?

Reply:
Yes, but note it says it can't prove exhaustiveness, it doesn't list the missing enum members.

I guess I can add a spec where all colors are covered but an else is still required... (though in that particular case I think all cases are actually covered, but I'm not sure we should do that).